stock_move_line_open_lot/tests/test_stock_move_line_open_lot.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2025 Ángel Rivas <angel.rivas@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.exceptions import UserError
+from odoo.tests.common import TransactionCase
+
+
+class TestStockMoveLineOpenLot(TransactionCase):
+    def setUp(self):
+        super().setUp()
+
+        self.product = self.env["product.product"].create(
+            {
+                "name": "Test Product",
+                "type": "product",
+                "tracking": "lot",
+            }
+        )
+
+        self.lot = self.env["stock.lot"].create(
+            {
+                "name": "LOT001",
+                "product_id": self.product.id,
+            }
+        )
+
+        self.stock_location = self.env["stock.location"].search(
+            [("name", "=", "Stock")], limit=1
+        )
+
+        self.stock_location_dest = self.env["stock.location"].search(
+            [("name", "=", "Customers")], limit=1
+        )
+
+        self.move = self.env["stock.move"].create(
+            {
+                "name": "Test Move",
+                "product_id": self.product.id,
+                "product_uom_qty": 1,
+                "product_uom": self.product.uom_id.id,
+                "location_id": self.stock_location.id,
+                "location_dest_id": self.stock_location_dest.id,
+            }
+        )
+
+        self.move_line = self.env["stock.move.line"].create(
+            {
+                "product_id": self.product.id,
+                "lot_id": self.lot.id,
+                "company_id": self.env.company.id,
+                "location_id": self.stock_location.id,
+                "location_dest_id": self.stock_location_dest.id,
+            }
+        )
+
+    def test_open_lot_form_action(self):
+        action = self.move_line.action_open_lot_form()
+        self.assertIsInstance(action, dict)
+        self.assertEqual(action.get("type"), "ir.actions.act_window")
+        self.assertEqual(action.get("res_model"), "stock.lot")
+        self.assertEqual(action.get("res_id"), self.lot.id)
+        self.assertEqual(action.get("view_mode"), "form")
+        self.assertEqual(self.product.tracking, "lot")
+
+    def test_open_lot_form_action_without_lot_id(self):
+        move_line_no_lot = self.env["stock.move.line"].create(
+            {
+                "product_id": self.product.id,
+                "company_id": self.env.company.id,
+                "location_id": self.stock_location.id,
+                "location_dest_id": self.stock_location_dest.id,
+                "move_id": self.move.id,
+                "lot_id": False,
+            }
+        )
+        with self.assertRaises(UserError):
+            move_line_no_lot.action_open_lot_form()
+
+    def test_compute_show_lot_button(self):
+        picking_type = self.env["stock.picking.type"].create(
```

Comment:
You can create the picking type in the set up function and then write it if needed in test_compute_show_lot_button and test_open_lot_form_action_with_popup